### PR TITLE
8299502: Usage of constructors of primitive wrapper classes should be avoided in javax.xml API docs

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Removed constructors of primitive wrapper classes (deprecated for removal) in _javax.xml.stream.XMLOutputFactory_ 

Replaced with Boolean static fields: Boolean.TRUE and Boolean.FALSE

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8299502](https://bugs.openjdk.org/browse/JDK-8299502): Usage of constructors of primitive wrapper classes should be avoided in javax.xml API docs

### Reviewers
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11872/head:pull/11872` \
`$ git checkout pull/11872`

Update a local copy of the PR: \
`$ git checkout pull/11872` \
`$ git pull https://git.openjdk.org/jdk pull/11872/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11872`

View PR using the GUI difftool: \
`$ git pr show -t 11872`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11872.diff">https://git.openjdk.org/jdk/pull/11872.diff</a>

</details>

### CODE REVIEW

File: src/java.xml/share/classes/javax/xml/stream/XMLOutputFactory.java [hunk 1]
```diff
@@ -56,7 +56,7 @@
  * <p>The following paragraphs describe the namespace and prefix repair algorithm:
  *
  * <p>The property can be set with the following code line:
- * {@code setProperty("javax.xml.stream.isRepairingNamespaces", new Boolean(true|false));}
+ * {@code setProperty("javax.xml.stream.isRepairingNamespaces", Boolean.TRUE|Boolean.FALSE);}
```
Comment (justin-curtis-lu) [inline src/java.xml/share/classes/javax/xml/stream/XMLOutputFactory.java:59]: From a code standpoint I don't see why the OR operation is needed. Is to emphasize that true or false can be passed into the parameter?

Comment (JoeWang-Java) [reply src/java.xml/share/classes/javax/xml/stream/XMLOutputFactory.java:59]: I think the later, that the value can be true or false. As an example, one value is fine, e.g.
  {@code setProperty("javax.xml.stream.isRepairingNamespaces", Boolean.TRUE);} 
  or 
  {@code setProperty("javax.xml.stream.isRepairingNamespaces", true);}
  let autoboxing handle it.

Comment (LanceAndersen) [reply src/java.xml/share/classes/javax/xml/stream/XMLOutputFactory.java:59]: The` true|false` should just be changed to just set the value to _true_ or _false_.

I think Boolean.TRUE is fine.   If the consensus is to use autoboxing in the example that is fine also.

Comment (justin-curtis-lu) [reply src/java.xml/share/classes/javax/xml/stream/XMLOutputFactory.java:59]: Thank you Joe and Lance, will replace it with a single value

**Review by JoeWang-Java — APPROVED** [review]

**Review by naotoj — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]

**Comment by altrisi** [comment]
Why not just `true` and `false`? Autoboxing handles that.

**Comment by justin-curtis-lu** [comment]
> Why not just `true` and `false`? Autoboxing handles that.

That was a topic that was talked about in [8299498](https://github.com/openjdk/jdk/pull/11912). It seemed that there was a consensus from a documentation standpoint that it was more clear to use the wrapper methods/fields (valueOf() or in this case Boolean.TRUE)

**Comment by altrisi** [comment]
I'd say in there it makes more sense given the method is used when you want a wrapper, and therefore explicitly converting it helps documentation, but here it's just setting a property, where I don't think it helps anything, just decreases readability as it seems like there may be a reason to call it explicitly with a wrapper and not a simple `true`/`false`.

**Comment by justin-curtis-lu** [comment]
> I'd say in there it makes more sense given the method is used when you want a wrapper, and therefore explicitly converting it helps documentation, but here it's just setting a property, where I don't think it helps anything, just decreases readability as it seems like there may be a reason to call it explicitly with a wrapper and not a simple `true`/`false`.

In `setProperty(String name, Object value)`, the method is expecting `value` to be an Object. Like the other methods, it is a matter of passing the wrapper class or letting the primitive auto box. Passing true or false would have auto boxed boolean to Boolean. 

The original statement before my change passed a Boolean, the consensus from the other PR was to pass the wrapper class, setProperty() is expecting `value` to be an Object, and the implementation of setProperty() casts `value` to be Boolean regardless. Those are the reasons why I picked the wrapper over the primitive. 

I think an argument can be made for either way, and I don't necessarily believe one way is right or wrong, hope that helped.

**Pull request closed without merging**